examples/jsm/nodes/ShaderNode.js

```diff
@@ -231,26 +233,46 @@ const ConvertType = function ( type ) {
 
 			}
 
+			if ( params.length === 1 && nodesCacheMap.has( params[ 0 ] ) ) {
+
+				return nodesCacheMap.get( params[ 0 ] );
+
+			}
+
 			const nodes = params.map( ( param ) => {
 
-				if ( cacheMap.has( param ) ) {
+				if ( paramsCacheMap.has( param ) ) {
 
-					return cacheMap.get( param );
+					return paramsCacheMap.get( param );
 
 				} else if ( param.isNode === true ) {
 
 					return param;
 
-				}
+				} else {
+
+					const node = new ConstNode( param );
+
+					if ( typeof param !== 'object' ) {
 
-				return new ConstNode( param );
+						paramsCacheMap.set( param, node );
+
+					}
+
+					return node;
 
 			} );
 
 			node = nodeObject( new ConvertNode( nodes.length === 1 ? nodes[ 0 ] : new JoinNode( nodes ), type ) );
 
 		}
 
+		if ( params.length === 1 && typeof params[ 0 ] !== 'object' ) {
+
+			nodesCacheMap.set( params[ 0 ], node );
```

Comment:
@LeviPesin The idea is to avoid dynamic cache storage because it's a global cache.

Reply:
Why there cannot be a static cache for parameters passed and a dynamic cache per type for caching values like `float(1)`, `float(0)`, and `int(0)`?

Reply:
We can have a `cacheLib = { float : new Map(), 'int' .. }` if necessary, but I would avoid dynamic strong storage here.

Reply:
> We can have a cacheLib = { float : new Map(), 'int' .. }

What would be the difference from the dynamic storage?

Reply:
Store a thousand different values and you never can reduce the cache again because it's a global strong reference, so let's I think that we keep it storage static just to common values.

Reply:
> Store a thousand different values and you never can reduce the cache again because it's a global strong reference

But that would be also true for any Map (which is not WeakMap), wouldn't it be? In both cases we have `numberOfTypes` maps which keys are primitive (and so their keys and values would not be GC'd in any case, I think...).

Reply:
for this reason we should fixed only these values https://github.com/mrdoob/three.js/pull/23635#discussion_r830668205

Reply:
So, you are suggesting to limit the `constNodesCacheMap` to only common float values and `true` and false, and to limit the `nodesCacheMap` depending on type - common floats for float, common ints for int, common uints for uint, and common bools for bool (and nothing for other types)?

Reply:
Is it better now?

Reply:
Yeess. Looks great! 👍